### ISSUE TITLE
common handling for rest requests

### ISSUE THREAD

**Issue by lipkau** [philips-software/powershell-cf-api#17]
I strongly recommend having a custom proxy function for `Invoke-WebRequest` or having an internal function for the module to use for all requests.
that way, you can dump all of the internals (baseurl, headers, pagination, etc) into one place and you have little refactoring in the public commands when you want to refactor internals or how calls are executed

Let me know if you are interested in a PR for this.

**Comment by marklindell** [comment]
I agree. This would also be a good place to implement retry logic on common HTTP failures due to connection issues.

**Comment by marklindell** [comment]
@lipkau Do you have any good examples of this I could steal, er, use?

**Comment by lipkau** [comment]
In my projects, I:
* create a private function for `Invoke-WebRequest` ([example](https://github.com/AtlassianPS/ConfluencePS/blob/master/ConfluencePS/Private/Invoke-WebRequest.ps1)) which makes mocking tests easier and abstracts the logic of Authorization headers as they differ between PSv5- and PSv6+
* create a public function `Invoke-Method` ([example](https://github.com/AtlassianPS/ConfluencePS/blob/master/ConfluencePS/Public/Invoke-Method.ps1)) which does the heavy lifting for:
  * pagination
  * type conversion (see #16)
  * error / response (so that you don't need a `try {} catch {}` block for handling 400+ status codes)
  * setting headers
  * setting the TLS version (code which you currently have repeated in multiple files)
